cura/BuildVolume.py

```diff
@@ -489,20 +489,21 @@ def _buildDisallowedAreaMesh(self, min_w: float, max_w: float, min_h: float, max
         if not self._disallowed_areas:
             return None
 
+        bounding_box = Polygon(numpy.array([[min_w, min_d], [min_w, max_d], [max_w, max_d], [max_w, min_d]], numpy.float32))
+
         mb = MeshBuilder()
         color = self._disallowed_area_color
         for polygon in self._disallowed_areas:
-            points = polygon.getPoints()
-            if len(points) == 0:
+            intersection = polygon.intersectionConvexHulls(bounding_box)
+
+            points = intersection.getPoints()[::-1]
```

Comment:
Think this notation is slightly more readable
```suggestion
            points = intersection.getPoints()
            points.reverse()
```

Reply:
It is, but `points` is a `numpy.array` so it does not have a `reverse` method. I'll see if there is some similar thing.